### DISCUSSION TITLE
How to make text inside TextEdit::multiline react on hovering?

### DISCUSSION THREAD

**Discussion by gravit22** [emilk/egui#2869] [Q&A]
Hello. 
How to make text inside TextEdit::multiline react on hovering?
For example, I want the popup appear with different variants of fixing grammatical mistakes of a hovered word inside the multiline edit, so I can write and see recommendations on hovering some words . How can I implement it with egui?
Thank you for your attention.

**Comment by rminderhoud** [comment]
This is in the docs for textedit: https://docs.rs/egui/0.21.0/egui/widgets/text_edit/struct.TextEdit.html

```
let response = ui.add(egui::TextEdit::singleline(&mut my_string));
if response.changed() {
    // …
}
if response.lost_focus() && ui.input(|i| i.key_pressed(egui::Key::Enter)) {
    // …
}
```

Check the response for hover state and then on next  frame before re-rendering the text edit make whatever adjustments you need.
